pkg/controller/elasticsearch/driver/upgrade_predicates.go

```diff
@@ -61,10 +97,20 @@ func applyPredicates(ctx PredicateContext, candidates []corev1.Pod, maxUnavailab
 			deletedPods = append(deletedPods, candidate)
 			allowedDeletions--
 			if allowedDeletions <= 0 {
-				break
+				break Loop
 			}
 		}
 	}
+
+	// If some predicates have failed print a summary of the failures to help
+	// the user to understand why.
+	if len(failedPredicates) > 0 {
+		log.Info(
+			"Nodes can't be upgraded",
+			"namespace", ctx.es.Namespace,
+			"es_name", ctx.es.Name,
+			"failedPredicates", groupByPredicates(failedPredicates))
```

Comment:
Nit: maybe stay in snake case to be consistent with the other fields: `failed_predicates`.